refiner-lib/src/refiner_inner.rs

```diff
@@ -856,6 +857,71 @@ fn build_transaction(
                     );
                     fill_with_submit_result(tx, result, &mut bloom)
                 }
+                TransactionKindTag::FtOnTransfer => {
+                    hash = virtual_receipt_id.0.into();
+                    tx = tx.hash(hash);
+
+                    if let Ok(args) = serde_json::from_slice::<NEP141FtOnTransferArgs>(&raw_input) {
+                        // For now, we use `engine_account_id` converted to the EVM address as the `from address` for both ETH and ERC-20 tokens.
+                        // Later, when the engine and ethconnector split is deployed on mainnet,
+                        // this will be changed to the address of ethconnector.
+                        let from_address =
+                            near_account_to_evm_address(engine_account_id.as_bytes());
+                        let to = determine_ft_on_transfer_to_address(
+                            execution_outcome,
+                            &args,
+                            storage,
+                            near_block,
+                            transaction_index,
+                        );
+                        let value = Wei::new(U256::from(args.amount.as_u128()));
```

Comment:
I don't think we want to set the value field in this way for the ERC-20 case. I think value is specifically meant for the amount of WEI, not arbitrary tokens.